clouddrift/transfer.py

```diff
@@ -11,6 +14,283 @@
+            Transfer function to apply to the input x. If a callable, it should accept
+            frequency array as input, see example below.
+            If ndarray, it must match the length of input x.
+        dt: float, optional
+            Time step size of the input time series, needed if transfer_func is callable.
+
+    Returns
+    -------
+        y: numpy ndarray
+            Output time series after applying the transfer function.
+
+    Examples
+    --------
+        Apply a low-pass filter transfer function to a random time series:
+        >>> import numpy as np
+        >>> from clouddrift.transfer import apply_transfer_function
+        >>> x = np.random.randn(100)
+        >>> transfer_func = lambda omega: 1 / (1 + (omega / (2 * np.pi * 0.1))**2)  # Low-pass filter
+        >>> dt = 1.0  # Time step needed for callable transfer functions
+        >>> y = apply_transfer_function(x, transfer_func, dt)
+
+        Directly provide as an array a transfer function which suppresses
+        the negative frequencies:
+        >>> import numpy as np
+        >>> from clouddrift.transfer import apply_transfer_function
+        >>> x = np.random.randn(100)
+        >>> transfer_func = np.ones(100)
+        >>> transfer_func[50:] = 0  # Suppress negative frequencies
+        >>> y = apply_transfer_function(x, transfer_func)
+
+        To a random wind stress, apply a slab wind transfer function at latitude 45 degrees,
+        with a friction coefficient of 0.1 times the Coriolis frequency and a boundary layer
+        depth of 50 meters:
+        >>> import numpy as np
+        >>> from clouddrift.sphere import coriolis_frequency, EARTH_DAY_SECONDS
+        >>> from clouddrift.transfer import apply_transfer_function, slab_wind_transfer
+        >>> tau = np.random.randn(100) + 1j * np.random.randn(100)  # Random wind stress
+        >>> cor_freq = coriolis_frequency(45) * EARTH_DAY_SECONDS  # Coriolis frequency at 45 degrees latitude in radians per day
+        >>> transfer_slab = lambda omega: slab_wind_transfer(omega, cor_freq[0] , friction=0.1 * cor_freq[0] / EARTH_DAY_SECONDS, bld=50)
+        >>> dt = 1.0  # Time step needed for callable transfer functions
+        >>> u = apply_transfer_function(tau, transfer_slab, dt)
+
+    Raises
+    ------
+        ValueError
+            If input x is not a one-dimensional array, or if transfer_func is callable
+            but dt is not provided, or if the shape of transfer_func does not match the length of x.
+    """
+
+    x = np.asarray(x)
+
+    if x.ndim != 1:
+        raise ValueError("Input x must be a 1D array.")
+    n = x.size
+
+    Xf = np.fft.fft(x)
+
+    if callable(transfer_func):
+        if dt is None:
+            raise ValueError(
+                "dt (time step size) must be provided when transfer_func is callable."
+            )
+        omega = 2 * np.pi * np.fft.fftfreq(n, dt)
+        G = transfer_func(omega)
+    else:
+        if dt is not None:
```

Comment:
you could add a simple test to cover this condition

Reply:
Ok I think I added a test to test if a warning is raised when dt is provided when transfer_func is an array. Is this waht you meant?